### DISCUSSION TITLE
osquery/fleet software and vulnerability overview missing ? Good old FleetDM days ...

### DISCUSSION THREAD

**Discussion by kawaiipantsu** [Security-Onion-Solutions/securityonion#11340] [2.4] [answered]
### Version

2.4.10

### Installation Method

Security Onion ISO image

### Description

other (please provide detail below)

### Installation Type

Standalone

### Location

on-prem with Internet access

### Hardware Specs

Exceeds minimum requirements

### CPU

6

### RAM

30

### Storage for /

2tb

### Storage for /nsm

2tb

### Network Traffic Collection

tap

### Network Traffic Speeds

1Gbps to 10Gbps

### Status

Yes, all services on all nodes are running OK

### Salt Status

No, there are no failures

### Logs

No, there are no additional clues

### Detail

Hi,

This is not a failure, bug or issue. This is a discussion about can we get the one feature back that i actually used FleetDM for :O)
That it showed all installed Software on the endpoints and that it could show vulnerability cve matches.

It's was a very neat way to get a good pretty overview on what software was installed and also if it had any associated vulnerabilities.
I tried to issue a "show install apps" via a live osquery but the Elastic agent just says "Apps not found"...

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to Security Onion 2.4.

**Comment by defensivedepth** [comment, marked as answer]
Check out the osquery table schema for how to query for apps for different platforms: https://osquery.io/schema/5.9.1/

For instance, for Windows: `select * from programs` yields the following (`programs` table schema is [here](https://osquery.io/schema/5.9.1/#programs)):

![image](https://github.com/Security-Onion-Solutions/securityonion/assets/954732/e6d32dcf-0c38-4793-ab80-6ab00f9debbf)


CVE's / vulnerabilities for installed apps would require a bit more work, it is not something that osquery does out of the box.
